### PR TITLE
Deprecate jupyterlab workspace

### PR DESCRIPTION
**Pull request by ZacWarham**
As per https://github.com/PrairieLearn/PrairieLearn/issues/10719#issuecomment-2380381046, the `prairielearn/workspace-jupyterlab` is no longer maintained. This PR is to provide some updates to the docs as suggested on Slack.

I separated the changes into individual commits so it is easier to determine the level of editorialising in the wording that may require future updates as opposed to being more generic (namely https://github.com/PrairieLearn/PrairieLearn/commit/92d501fc139feb45d0e190efc0308debd2ab165d).

Feel free to provide any specific wording changes or push an additional commit to fix.

### CODE REVIEW

File: docs/workspaces/index.md [hunk 1]
```diff
@@ -221,6 +221,16 @@ If a file name appears in multiple locations, the following precedence takes eff
...
 
+## Maintained workspace images
+
+PrairieLearn provides and maintains the following workspace images:
+
+- [An Ubuntu 24.04 desktop](https://github.com/PrairieLearn/PrairieLearn/tree/master/workspaces/desktop/)
+- [JupyterLab with Python 3.11](https://github.com/PrairieLearn/PrairieLearn/tree/master/workspaces/jupyterlab-python/)
+- [RStudio with R version 4.4](https://github.com/PrairieLearn/PrairieLearn/tree/master/workspaces/rstudio/)
+- [VS Code with Python 3.10](https://github.com/PrairieLearn/PrairieLearn/tree/master/workspaces/vscode-python/)
+- [Terminal emulator based on xterm.js](https://github.com/PrairieLearn/PrairieLearn/tree/master/workspaces/xtermjs/)
```
Comment (nwalters512) [inline docs/workspaces/index.md:232]: Let's use the workspace image names here, e.g.:

- [`prairielearn/workspace-jupyterlab-python`](https://github.com/PrairieLearn/PrairieLearn/tree/master/workspaces/jupyterlab-python/): JupyterLab with Python 3.11

File: workspaces/jupyterlab/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## Deprecated workspace image
+
+This workspace image is no longer maintained by PrairieLearn and is not recommended for use or meant as a foundation for other images. Instead, please see [jupyterlab-python](../jupyterlab-python/).
```
Comment (nwalters512) [inline workspaces/jupyterlab/README.md:3]:
```suggestion
This workspace image is no longer maintained by PrairieLearn and is not recommended for use or meant as a foundation for other images. Instead, please see [`prairielearn/workspace-jupyterlab-python`](../jupyterlab-python/).
```

File: workspaces/README.md [hunk 3]
```diff
@@ -1,4 +1,14 @@
...
+## PrairieLearn workspace images
 
 Within this directory are a set of docker containers that are intended to be run inside of workspaces.
 These docker containers may be modified as needed on a per-course basis.
+
+#### Maintained images
+
+The following workspace images are maintained by PrairieLearn:
+
+- [An Ubuntu 24.04 desktop](./desktop/)
```
Comment (nwalters512) [inline workspaces/README.md:10]: Ditto, let's use workspace image names.

File: workspaces/README.md [hunk 4]
```diff
@@ -1,4 +1,14 @@
-## PrairieLearn Workspace Images
+## PrairieLearn workspace images
 
 Within this directory are a set of docker containers that are intended to be run inside of workspaces.
 These docker containers may be modified as needed on a per-course basis.
```
Comment (nwalters512) [inline workspaces/README.md:4]: Nits, as long as you are changing things:

```suggestion
Within this directory are a set of Docker images that are intended to be run inside of workspaces.
These Docker images may be modified as needed on a per-course basis.
```

File: docs/workspaces/index.md (hunk 1 above)
Comment (ZacWarham) [reply docs/workspaces/index.md:232]: Just to clarify @nwalters512, would you like me to replace `JupyterLab with Python 3.11` with `prairielearn/workspace-jupyterlab-python` or keep both the workspace image name and the editorialised name?

File: docs/workspaces/index.md [hunk 5]
```diff
@@ -221,6 +221,16 @@ If a file name appears in multiple locations, the following precedence takes eff
 
 - Files in the `workspace/` directory are considered last.
 
+## Maintained workspace images
+
+PrairieLearn provides and maintains the following workspace images:
+
+- [prairielearn/workspace-desktop](https://github.com/PrairieLearn/PrairieLearn/tree/master/workspaces/desktop/): An Ubuntu 24.04 desktop
```
Comment (nwalters512) [inline docs/workspaces/index.md:228]: Please use backticks for image names like in the example I gave:

```suggestion
- [`prairielearn/workspace-desktop`](https://github.com/PrairieLearn/PrairieLearn/tree/master/workspaces/desktop/): An Ubuntu 24.04 desktop
```

Ditto for images below and in other files.

Comment (ZacWarham) [reply docs/workspaces/index.md:228]: Ah yes, sorry about that

File: workspaces/README.md [hunk 6]
```diff
@@ -1,4 +1,14 @@
-## PrairieLearn Workspace Images
+## PrairieLearn workspace images
 
-Within this directory are a set of docker containers that are intended to be run inside of workspaces.
-These docker containers may be modified as needed on a per-course basis.
+Within this directory are a set of Docker images that are intended to be run inside of workspaces.
+These Docker images may be modified as needed on a per-course basis.
```
Comment (nwalters512) [inline workspaces/README.md:4]: Final cleanup nits (I know this is unrelated to your changes, but as long as we're improving things...):
- These images aren't run _inside of_ workspaces, but rather are themselves the workspace images.
- Given that this file doesn't (and IMO shouldn't) contain comprehensive information about _how_ to actually modify the images, I'm inclined to say we should just drop the second sentence entirely.

```suggestion
This directory contains source files for Docker images that are intended to be used as [PrairieLearn workspaces](https://prairielearn.readthedocs.io/en/latest/workspaces/).
```

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**